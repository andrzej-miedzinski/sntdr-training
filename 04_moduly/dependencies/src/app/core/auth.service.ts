import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  login() {
    console.log('Użytkownik zalogowany.');
  }

  logout() {
    console.log('Użytkownik wylogowany.');
  }
}