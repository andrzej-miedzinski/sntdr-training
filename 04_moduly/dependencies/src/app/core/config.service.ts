import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  getConfig() {
    return { apiUrl: 'https://api.example.com' };
  }
}