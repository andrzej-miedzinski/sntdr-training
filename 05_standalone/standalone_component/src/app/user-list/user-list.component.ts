import { Component } from '@angular/core';
import { UserService } from '../user.service';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-user-list',
  standalone: true,
  imports: [CommonModule],
  providers: [UserService], // Bezpośrednie dostarczanie usługi
  template: `
    <ul>
      <li *ngFor="let user of users">{{ user.name }}</li>
    </ul>
  `
})
export class UserListComponent {
  users: any[] = [];

  constructor(private userService: UserService) {
    this.users = this.userService.getUsers();
  }
}