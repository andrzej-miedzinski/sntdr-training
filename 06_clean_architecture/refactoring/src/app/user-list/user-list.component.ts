import { Component } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-user-list',
  template: `
    <ul>
      <li *ngFor="let user of users">{{ user.name }} - {{ user.email }}</li>
    </ul>
  `
})
export class UserListComponent {
  users: any[] | undefined = [];

  constructor(private http: HttpClient) {}

  async ngOnInit() {
    this.users = await this.http.get<any[]>('https://api.example.com/users').toPromise();
  }
}