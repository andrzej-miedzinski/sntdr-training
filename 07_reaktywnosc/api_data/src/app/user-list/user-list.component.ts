import { Component } from '@angular/core';
import { UserService } from '../user.service';
import { Observable } from 'rxjs';

@Component({
  selector: 'app-user-list',
  template: `
    <ul>
      <li *ngFor="let user of users$ | async">{{ user.fullName }}</li>
    </ul>
  `
})
export class UserListComponent {
  users$ : Observable<any[]>;
  

  constructor(private userService: UserService) {
    this.users$ = this.userService.getUsers();
  }
}