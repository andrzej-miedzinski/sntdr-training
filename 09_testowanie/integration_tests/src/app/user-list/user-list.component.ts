import { Component, OnInit } from '@angular/core';
import { UserService } from '../user.service';

@Component({
  selector: 'app-user-list',
  template: `
    <ul>
      <li *ngFor="let user of users">{{ user.name }}</li>
    </ul>
  `
})
export class UserListComponent implements OnInit {
  users: any[] = [];

  constructor(private userService: UserService) {}

  ngOnInit() {
    this.users = this.userService.getUsers();
  }
}