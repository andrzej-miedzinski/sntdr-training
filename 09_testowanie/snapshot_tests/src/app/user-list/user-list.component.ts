import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-user-list',
  template: `
    <ul>
      <li *ngFor="let user of users">{{ user.name }}</li>
    </ul>
  `
})
export class UserListComponent {
  @Input() users: { name: string }[] = [];
}