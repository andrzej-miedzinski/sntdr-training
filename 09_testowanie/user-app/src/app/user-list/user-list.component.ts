import { Component, OnInit } from '@angular/core';
import { UserService } from '../services/user.service';

@Component({
  selector: 'app-user-list',
  template: `
    <h2>Lista użytkowników</h2>
    <ul>
      <li *ngFor="let user of users">{{ user.name }}</li>
    </ul>
  `
})
export class UserListComponent implements OnInit {
  users: any[] = [];

  constructor(private userService: UserService) {}

  ngOnInit() {
    this.userService.getUsers().subscribe(data => {
      this.users = data;
    });
  }
}