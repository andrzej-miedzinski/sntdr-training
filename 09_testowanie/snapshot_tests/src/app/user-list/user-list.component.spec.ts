import { TestBed, ComponentFixture } from '@angular/core/testing';
import { UserListComponent } from './user-list.component';

describe('UserListComponent', () => {
  let component: UserListComponent;
  let fixture: ComponentFixture<UserListComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [UserListComponent]
    });
    fixture = TestBed.createComponent(UserListComponent);
    component = fixture.componentInstance;
    component.users = [{ name: 'John' }, { name: 'Jane' }];
    fixture.detectChanges(); // Renderuje komponent
  });

  it('powinien pasowaÄ‡ do snapshotu', () => {
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot();
  });
});