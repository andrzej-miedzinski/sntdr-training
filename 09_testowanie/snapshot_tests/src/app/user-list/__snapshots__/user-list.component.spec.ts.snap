// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListComponent powinien pasowaÄ‡ do snapshotu 1`] = `
"<div id="root0" ng-version="18.2.13"><ul><li>John</li><li>Jane</li><!--bindings={
  "ng-reflect-ng-for-of": "[object Object],[object Object"
}--></ul></div>"
`;
